From 78863773cf46ac6777915ee41ee86838d297249f Mon Sep 17 00:00:00 2001
From: Anu Engineer <aengineer@apache.org>
Date: Tue, 7 Jun 2016 10:18:05 -0700
Subject: [PATCH 1603/2848] HDFS-10476. DiskBalancer: Plan command output
 directory should be a sub-directory. Contributed
 by Anu Engineer.

(cherry picked from commit 9f6e1d61f04c18ec7382e3469086ce484d0f15d9)

Change-Id: I5e6ba4f052980a0aa9e561142b26079d6cec7183
---
 .../hdfs/server/diskbalancer/command/Command.java  |    4 ++--
 .../org/apache/hadoop/hdfs/tools/DiskBalancer.java |    4 ++++
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/diskbalancer/command/Command.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/diskbalancer/command/Command.java
index db1c0e2..704e79e 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/diskbalancer/command/Command.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/diskbalancer/command/Command.java
@@ -161,10 +161,10 @@ protected void setOutputPath(String path) throws IOException {
       if (getClusterURI().getScheme().startsWith("file")) {
         diskBalancerLogs = new Path(
             System.getProperty("user.dir") + DEFAULT_LOG_DIR.toString() +
-                format.format(now));
+                Path.SEPARATOR + format.format(now));
       } else {
         diskBalancerLogs = new Path(DEFAULT_LOG_DIR.toString() +
-            format.format(now));
+            Path.SEPARATOR + format.format(now));
       }
     } else {
       diskBalancerLogs = new Path(path);
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/DiskBalancer.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/DiskBalancer.java
index 87fbf4a..b0bd07e 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/DiskBalancer.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/DiskBalancer.java
@@ -203,6 +203,10 @@ private void addPlanCommands(Options opt) {
     Option help =
         new Option(HELP, true, "Help about a command or this message");
     opt.addOption(help);
+
+    Option verbose = new Option(VERBOSE, "Print out the summary of the plan");
+    opt.addOption(verbose);
+
   }
 
   /**
-- 
1.7.9.5

